### DISCUSSION TITLE
Restarting Vaultwarden process with container signal

### DISCUSSION THREAD

**Discussion by tommyalatalo** [dani-garcia/vaultwarden#2600] [Q&A]
I need to restart my vaultwarden process inside the container due to updating the database credentials dynamically in my configuration.
I was expecting to be able to send the container `SIGHUP` to restart the process, and it seems that vaultwarden gets the signal and does something, but it's not recovering properly. What is `SIGHUP` configured to do in the container, and is there another signal that should be used to restart the process without restarting the entire container?

**Comment by hfath** [comment]
From a cursory look, it does not seem like vaultwarden has any signal handling. It is generally not adhering well to unixy conventions.

I would like to set up log wrapping with newsyslog(8), but sending HUP will just terminate bitwarden...

**Comment by BlackDex** [comment]
Vaultwarden currently does not support all types of signals.
SIGHUP in this case is not supported. Mostly because for Vaultwarden to reload the full config we need to make some larger changes on how to handle the config and environment variables for example. And stuff like port changes, etc..

Best way to _reload_ is to just restart Vaultwarden. It has the same effect.

Btw, it does nicely response to ctrl+c

**Reply by hfath** [reply]
Unfortunately, the *BSD newsyslog(8) (see [newsyslog,conf](https://www.freebsd.org/cgi/man.cgi?query=newsyslog.conf) ) can nudge, but not restart daemons.

I looked into logging to syslogd, but that hard-codes the facility to LOG_USER, AFAICS...

**Reply by hfath** [reply]
FTR, I just committed a security/vaultwarden package for NetBSD pkgsrc. I might just patch the facility...
